contracts/hub/Base.sol

```diff
@@ -56,9 +56,14 @@ abstract contract Base is ReentrancyGuard, IHub, RescueBase {
         address hook_,
         bool approve_
     ) external virtual onlyOwner {
-        hook__ = IHook(hook_);
-        if (approve_)
+        if (approve_) {
+            // remove the approval from the old hook
+            SafeTransferLib.safeApprove(ERC20(token), address(hook__), 0);
+
             SafeTransferLib.safeApprove(ERC20(token), hook_, type(uint256).max);
+        }
+        hook__ = IHook(hook_);
+
```

Comment:
for old hook, we should revoke based on if non zero approval is present on token contract instead of relying on approve_ bool param.